app/views/components/_timeout_modal.html.erb

```diff
@@ -0,0 +1,110 @@
+<div class="margin-y-3">
+  <a
+    id="button-toggle-timeout"
+    href="#session-timeout-modal"
+    class="hidden"
+    aria-controls="session-timeout-modal"
+    data-open-modal
+    ></a
+  >
+  <div
+    class="usa-modal"
+    id="session-timeout-modal"
+    aria-labelledby="modal-1-heading"
+    aria-describedby="modal-1-description"
+    data-force-action
+  >
+    <div class="usa-modal__content">
+      <div class="usa-modal__main">
+        <h2 class="usa-modal__heading" id="modal-1-heading">
+          Are you still using Touchpoints?
+        </h2>
+        <div class="usa-prose">
+          <p id="modal-1-description">
+            For security reasons,
+            your session will timeout in 5 minutes due to inactivity.
```

Comment:
The timeout here isn't strictly an inactivity timeout, is it? It resets when you reload the page but not for ajax calls. So maybe it shouldn't say "due to inactivity".

Reply:
correct.

on Data Collections, there are measures to maintain a heartbeat and activity for ajax requests.

This PR does not address ajax, nor extend that pattern.